src/Stratis.SmartContracts.CLR/SCL/Operations.cs

```diff
@@ -1,9 +1,54 @@
 ﻿using System;
+using System.Linq;
 
 namespace Stratis.SCL.Base
 {
     public static class Operations
     {
         public static void Noop() { }
+
+        /// <summary>
+        /// Unflattens an array of size N * <paramref name="subArrayLength"/> into an array of N sub-arrays each of length <paramref name="subArrayLength"/>.
+        /// </summary>
+        /// <typeparam name="T">The type of the array elements.</typeparam>
+        /// <param name="array">The array to convert.</param>
+        /// <param name="subArrayLength">The length of the sub-arrays to create.</param>
+        /// <returns>An array of N sub-arrays each of length <paramref name="subArrayLength"/>.</returns>
+        /// <remarks>The value of N is implied by the length of the input array and the value of <paramref name="subArrayLength"/>.</remarks>
+        public static T[][] UnflattenArray<T>(T[] array, int subArrayLength) where T : struct
+        {
+            int cnt = array.Length / subArrayLength;
+
+            if (array.Length != subArrayLength * cnt)
+                return null;
+
+            var buffer = new T[cnt][];
+            for (int i = 0; i < cnt; i++)
+            {
+                buffer[i] = new T[subArrayLength];
+                Array.Copy(array, i * subArrayLength, buffer[i], 0, subArrayLength);
+            }
+
+            return buffer;
+        }
+
+        /// <summary>
+        /// Flattens an array of N sub-arrays each of length <paramref name="subArrayLength"/> into an array of size N * <paramref name="subArrayLength"/>.
+        /// </summary>
+        /// <typeparam name="T">The type of the array elements.</typeparam>
+        /// <param name="array">The array to convert.</param>
+        /// <param name="subArrayLength">The length of the sub-arrays.</param>
+        /// <returns>An array of size N * <paramref name="subArrayLength"/></returns>
+        public static T[] FlattenArray<T>(T[][] array, int subArrayLength) where T : struct
+        {
+            if (array == null || array.Any(x => x == null || x.Length != subArrayLength))
```

Comment:
It is needed to validate parameters ? Because below logic will already fail if array is null and contains null items. 

These are going to cost gas so code should to be efficient. You can try catch whole logic and return null. Secondly below code already iterate values so you may validate items there. Imo if it will fail already so no need to check there even.

Reply:
I think letting an error occur within try / catch is expensive.

Reply:
Done.

Reply:
Could we perhaps make this private for now? Otherwise there's 3 decisions about the SCL API in this PR.

I'm fine with the implementation. I think we could do a full review of additional optimizations before SCL is published.

Reply:
@rowandh , I've removed flatten / unflatten in favour of using the `ContractPrimitiveSerializer.ToArray`.